azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/model/implementation/CreatableUpdatableImpl.java

```diff
@@ -18,217 +20,166 @@
-    public FluentModelImplT update() {
-        return (FluentModelImplT) this;
+    @Override
+    public boolean isHot() {
+        // createResourceAsync & updateResourceAsync returns cold observable since Retrofit Http
+        // request APIs (POST, PUT, PATCH ..) returns cold observable
+        return false;
     }
 
-    /**
-     * Default implementation of createAsync().
-     *
-     * @return the observable that emit the created resource.
-     */
     @Override
     public Observable<Indexable> createAsync() {
         return this.executeTaskGroupAsyncStreaming();
     }
 
-    /**
-     * Default implementation to create the resource asynchronously.
-     *
-     * @param callback the callback to handle success and failure
-     * @return a handle to cancel the request
-     */
+    @Override
     public ServiceCall<FluentModelT> createAsync(final ServiceCallback<FluentModelT> callback) {
         return observableToFuture(Utils.<FluentModelT>rootResource(createAsync()), callback);
     }
 
-    /**
-     * Default implementation of create().
-     *
-     * @return the created resource
-     */
+    @Override
     @SuppressWarnings("unchecked")
     public FluentModelT create() {
         return Utils.<FluentModelT>rootResource(createAsync()).toBlocking().single();
     }
 
     /**
-     * Default implementation of applyAsync().
+     * Default implementation of {@link Updatable#update()}.
      *
-     * @return the observable that emit the updated resource.
+     * @return the first stage of model update
      */
+    @SuppressWarnings("unchecked")
+    public FluentModelImplT update() {
+        return (FluentModelImplT) this;
+    }
+
+
     @Override
     public Observable<FluentModelT> applyAsync() {
         return this.executeTaskGroupAsync();
     }
 
-    /**
-     * Default implementation to update the resource asynchronously.
-     *
-     * @param callback the callback to handle success and failure
-     * @return a handle to cancel the request
-     */
     @Override
     public ServiceCall<FluentModelT> applyAsync(ServiceCallback<FluentModelT> callback) {
         return observableToFuture(applyAsync(), callback);
     }
 
-    /**
-     * Default implementation of apply().
-     *
-     * @return the created resource
-     */
     @Override
     public FluentModelT apply() {
         return applyAsync().toBlocking().last();
     }
 
-    /**
-     * This is the default implementation of 'updateResourceAsync', it simply calls createResourceAsync()
-     * since for most of the resource both create and update are handled by the same API call
-     * (CreateOrUpdate). A resource that uses different API call for update should override updateResourceAsync.
-     *
-     * @return the updated resource.
-     */
     @Override
     public Observable<FluentModelT> updateResourceAsync() {
         return this.createResourceAsync();
     }
 
+    @Override
+    public abstract Observable<FluentModelT> createResourceAsync();
```

Comment:
I don't think we are doing this anywhere else - I'm adding base impls to collection interfaces like `SupportsListingByXXX` to support the 3 async overloads - this would be a lot of work over there to keep it consistent